src/artifact.c

```diff
@@ -9839,6 +9850,47 @@ arti_invoke(obj)
 				do_bloodletter(obj);
 			}
 
+		break;
+		case WAVES_DARKNESS:
+			if (!uwep || uwep != obj){
+				You_feel("that you should be wielding %s.", the(xname(obj)));
+				obj->age = monstermoves;
+				return MOVE_CANCELLED;
+			}
+#define astel_blast(x, y, size) explode(x, y, AD_MAGM, WAND_CLASS, (d(1, 12) + uwep->spe) * ((Double_spell_size) ? 3 : 2) / 2, EXPL_CYAN, size + !!Double_spell_size)
```

Comment:
Add an isok check to x and y before calling this as the code below can potentially call it off the map and I think calling explode off of the map is still BAD.

Reply:
👍